### PR TITLE
metadata PR more fixes

### PR DESCRIPTION
**Pull request by kolharsam**
#### Work done

- action relationships (fixes on the form and also the request, was able to add, sometimes remove as well (for object relationships, but it wasn’t working for array relationships for some reason will check this once again))

- action permissions - (migrations were incorrectly generated here)

- event invocations for manual trigger invocation. I trust it should work since, this is the same one that’s used for the redeliver case, so yeah, fairly confident it should work here too)

- views, tables, functions couldn't update comments from the modify section earlier. Now it is fixed.

- track table/view/function from RawSQL is now fixed. Although not in the manner that may be correct/preferred approach.

#### Checked Off

 ~~- allowlist~~ (things are looking gooood here! 😄 )

 ~~- track table/fn from RawSQL~~

#### TODO

 - custom functions un-tracking (possible server issue again here, because the response says that the custom function could not be located in the schema cache) (see below for the request and the error)

#### Issues

- possible bug while trying to alter column related details from the console. So running something like this request:
```json
{
    "type": "bulk",
    "source": "pg-11",
    "args": [{
        "type": "run_sql",
        "args": {
            "sql": "ALTER TABLE public.new_tab alter column first_name type text;"
        }
    }, {
        "type": "run_sql",
        "args": {
            "sql": "COMMENT on column public.new_tab.last_name is 'column comment here'"
        }
    }]
}
```

 used while modifying the comment or any other detail about a particular column. The server never responds it's just stuck in a loop or some locks aren't released. So yeah, that is causing some issues. 

Update: Upon some digging, I see that it's the `comment` statement that is causing some issues.

- Table Permissions Icon has to be the correct one here. I have given permissions on all columns and yet the icon is different
image

- Custom Function un-tracking request 
```json
{
            "type": "pg_untrack_function",
            "args": {
                "name": "search_articles",
                "schema": "public",
                "source": "pg-11"
            }
        }
```

Response : 
```json
{
    "path": "$[0]",
    "error": "function not found in cache \"search_articles\"",
    "code": "not-exists"
}
```

### CODE REVIEW

File: console/src/components/Services/Actions/ServerIO.js [hunk 1]
```diff
@@ -479,17 +478,17 @@ export const removeActionRel = (relName, source, typename, successCb) => (
...
   const migration = new Migration();
   migration.add(customTypesQueryUp, customTypesQueryDown);
 
-  const migrationName = 'remove_action_rel'; // TODO: better migration name
-  const requestMsg = 'Removing relationship...';
+  const migrationName = `remove_action_relationship_${relName}_from_${typename}`;
+  const requestMsg = `Removing relationship ${relName}...`;
   const successMsg = 'Relationship removed successfully';
-  const errorMsg = 'Removing relationship failed';
+  const errorMsg = `Failed to remove the relationship: ${relName}`;
```
Comment (beerose) [inline console/src/components/Services/Actions/ServerIO.js]:
```suggestion
  const errorMsg = `Failed to remove the relationship: "${relName}"`;
```

File: console/src/components/Services/Actions/ServerIO.js [hunk 2]
```diff
@@ -525,10 +524,13 @@ export const saveActionPermission = (successCb, errorCb) => (
...
 
-  const migrationName = 'save_action_perm';
+  const { role, newRole } = permissionEdit;
+  const roleName = (newRole || role).trim();
+
+  const migrationName = `save_action_permission_${currentAction}_${roleName}`;
   const requestMsg = 'Saving permission...';
   const successMsg = 'Permission saved successfully';
-  const errorMsg = 'Saving permission failed';
+  const errorMsg = `Failed to save permissions for role ${roleName}`;
```
Comment (beerose) [inline console/src/components/Services/Actions/ServerIO.js]:
```suggestion
  const errorMsg = `Failed to save permissions for role "${roleName}"`;
```

File: console/src/components/Services/Events/EventTriggers/InvokeManualTrigger/InvokeManualTriggerAction.js [hunk 3]
```diff
@@ -14,30 +13,23 @@ const FETCH_EVENT_STATUS_SUCCESS =
...
-      payload: {
-        new: args,
-        old: null,
-      },
-    },
-  };
-  dispatch({ type: INVOKING_EVENT_TRIGGER });
+  // payload strictly needs to be a JSON
+  const payload = JSON.parse(JSON.stringify({ row: args }));
+  const query = invokeManualTriggerQuery({ name, source, payload }, source);
```
Comment (beerose) [inline console/src/components/Services/Events/EventTriggers/InvokeManualTrigger/InvokeManualTriggerAction.js]:
```suggestion
  const query = invokeManualTriggerQuery({ name, source, { row: args }}, source);
```

File: console/src/components/Services/Events/EventTriggers/InvokeManualTrigger/InvokeManualTriggerAction.js [hunk 4]
```diff
@@ -14,30 +13,23 @@ const FETCH_EVENT_STATUS_SUCCESS =
...
-    args: {
-      name: name,
-      payload: {
-        new: args,
-        old: null,
-      },
-    },
-  };
-  dispatch({ type: INVOKING_EVENT_TRIGGER });
+  // payload strictly needs to be a JSON
```
Comment (beerose) [inline console/src/components/Services/Events/EventTriggers/InvokeManualTrigger/InvokeManualTriggerAction.js]:
```suggestion
```

File: console/src/metadata/queryUtils.ts [hunk 5]
```diff
@@ -92,6 +92,12 @@ type MetadataQueryArgs = {
   [key: string]: any;
 };
 
+export type InvokeManualTriggerArgs = {
+  name: string;
+  source: string;
+  payload: JSON;
```
Comment (beerose) [inline console/src/metadata/queryUtils.ts]:
```suggestion
  payload: Record<string, any>;
```

File: console/src/metadata/types.ts [hunk 6]
```diff
@@ -740,6 +740,8 @@ export interface CustomTypeObjectRelationship {
   field_mapping: {
     [ObjectFieldName: string]: string;
   };
+  /** Source name, where remote_table exists */
+  source?: 'default' | string;
```
Comment (beerose) [inline console/src/metadata/types.ts]:
```suggestion
  source?: string;
```

File: console/src/utils/migration/utils.ts [hunk 7]
```diff
@@ -32,11 +32,12 @@ export interface SchemaType {
 }
 
 export interface OldColumnType {
-  data_type_name: string;
   data_type: string;
   column_default: string | null;
   comment: string | null;
+  udt_name?: string;
```
Comment (beerose) [inline console/src/utils/migration/utils.ts]:
```suggestion
```

File: console/src/utils/migration/utils.ts [hunk 8]
```diff
@@ -53,8 +54,9 @@ const parseNewCol = (newColumn: NewColumnType) => ({
   customFieldName: (newColumn.customFieldName || '').trim(),
 });
 
+// FIXME: I could never see the property udt_name, need to check this again
 const parseOldColumns = (oldColumn: OldColumnType) => ({
-  originalColType: oldColumn.data_type_name,
+  originalColType: oldColumn.udt_name || oldColumn.data_type_name,
```
Comment (beerose) [inline console/src/utils/migration/utils.ts]:
```suggestion
  originalColType: oldColumn.data_type_name,
```

File: console/src/utils/migration/utils.ts [hunk 9]
```diff
@@ -114,7 +116,7 @@ export const getColumnUpdateMigration = (
 
   const migration = new Migration();
 
-  if (originalColType !== colType) {
+  if (originalColType !== colType && originalColType?.length !== 0) {
```
Comment (beerose) [inline console/src/utils/migration/utils.ts]:
```suggestion
  if (originalColType !== colType) {
```

File: console/src/components/Services/Events/EventTriggers/InvokeManualTrigger/InvokeManualTriggerAction.js [hunk 10]
```diff
@@ -14,30 +13,23 @@ const FETCH_EVENT_STATUS_SUCCESS =
...
   const options = {
     method: 'POST',
     headers: dataHeaders(getState),
-    body: JSON.stringify(manualTriggerObj),
+    body: JSON.stringify(query),
   };
+
   return dispatch(requestAction(url, options))
     .then(data => {
+      alert(data);
```
Comment (beerose) [inline console/src/components/Services/Events/EventTriggers/InvokeManualTrigger/InvokeManualTriggerAction.js]:
```suggestion
```

**Review by beerose — APPROVED** [review]

File: console/src/utils/migration/utils.ts [hunk 11]
```diff
@@ -53,8 +54,9 @@ const parseNewCol = (newColumn: NewColumnType) => ({
   customFieldName: (newColumn.customFieldName || '').trim(),
 });
 
+// FIXME: I could never see the property udt_name, need to check this again
```
Comment (beerose) [inline console/src/utils/migration/utils.ts]:
```suggestion
```

File: console/src/components/Services/Data/RawSQL/Actions.js [hunk 12]
```diff
@@ -36,63 +36,21 @@ const MODAL_OPEN = 'EditItem/MODAL_OPEN';
 const modalOpen = () => ({ type: MODAL_OPEN });
 const modalClose = () => ({ type: MODAL_CLOSE });
 
-const executeSQLCallback = (isMigration, migrationName, url, sql) => (
```
Comment (beerose) [inline console/src/components/Services/Data/RawSQL/Actions.js:39]: to do

File: console/src/dataSources/services/postgresql/sqlUtils.ts [hunk 13]
```diff
@@ -666,21 +666,36 @@ export const getSetCommentSql = (
   on: 'column' | 'table' | string,
   tableName: string,
   schemaName: string,
-  columnName: string,
-  comment: string | null
-) => `
+  comment: string | null,
```
Comment (beerose) [inline console/src/dataSources/services/postgresql/sqlUtils.ts:669]: to do

**Comment by kolharsam** [comment]
Some of the bugs and other TODOs are being tracked on #45

**Pull request merged**